packages/js/email-editor/development.md

```diff
@@ -8,8 +8,86 @@ The most efficient way to develop the Email Editor is by using the WooCommerce p
 pnpm --filter='@woocommerce/plugin-woocommerce' watch:build:admin
 ```
 
+---
+
 ## Running Tests
 
-TBD
+### JavaScript Component Tests
+
+To run component tests in the JS package:
+
+```bash
+pnpm test:js
+```
+
+To run a specific test file:
+
+```bash
+pnpm test:js -- src/components/my-component/test/my-component.spec.tsx
+```
+
+We use [Jest](https://jestjs.io/) with `@testing-library/react`. These are **component tests**, not strict unit tests, and include mocked dependencies.
+
+#### Guidelines for Writing Component Tests
+
+- Use `should` prefix in test names (e.g., `should render the modal`).
+- Avoid testing implementation details. Prefer visible DOM assertions.
+- Use [jest.fn()](https://jestjs.io/docs/mock-functions) and [jest.mock()](https://jestjs.io/docs/manual-mocks) for mocking.
+- Always mock required dependencies **before** importing the tested component.
+- Avoid writing tests for components that are simple wrappers of 3rd-party libraries (e.g., just rendering a WordPress component without added logic).
+- Prefer reusable mocks for commonly used packages (e.g., `@wordpress/data`, `@wordpress/components`). Create shared mock setup files when possible.
+- Use descriptive `data-testid` attributes when mocking components (e.g., `data-testid="modal"`).
```

Comment:
I struggle to understand what this sentence means. Could be because I'm not familiar with jest. Could you please explain? We could perhaps improve this sentence.